message_app_client/src/components/conversation_panel/ConversationCard.js

```diff
@@ -11,14 +11,15 @@ function ConversationCard(props){
     // props.conversationID: int
 
     const [otherParticipantStatus, setOtherParticipantStatus] = useState(props.otherParticipantStatus);
-    const { conversationID, setConversationID } = useContext(ConversationDataContext)
+    const { conversationID, setConversationID, setOtherParticipantName } = useContext(ConversationDataContext);
 
     const joinConversation = () => {
         // If a user try to join a conversation they are already in, do nothing
         if (props.conversationID === conversationID) return;
 
         let oldConversationID = conversationID;
         setConversationID(props.conversationID)
+        setOtherParticipantName(props.conversationTitle)
```

Comment:
I am wondering if we can totally remove the use of `props` here if we are already using `context` to pass data around?

Reply:
Not in this component. The `ConversationDataContext` stores the data of the conversation that the user is currently in, while `ConversationCard` contains the data for each conversation. When the user chooses a conversation, the states inside `ConversationDataContext` will be updated based on the `ConversationCard`'s data.

Reply:
Gotcha